src/Autocomplete.tsx

```diff
@@ -0,0 +1,83 @@
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+
+import React, { useState, useCallback, useMemo } from 'react';
+import { Person } from './types/Person';
+
+interface Props {
+  people: Person[]
+  delay?: number
+  onSelected: (value: Person | null) => void
+}
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  delay,
+  onSelected,
+}) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), [delay]);
+
+  const filteredPeople = useMemo(() => {
+    const normalizedQuery = appliedQuery.toLowerCase().trim();
+
+    return people.filter(
+      person => person.name.toLowerCase().includes(normalizedQuery),
+    );
+  }, [appliedQuery, people]);
+
+  const handleSelect = (person: Person) => {
+    setQuery(person.name);
+    onSelected(person);
+    setIsDropdownOpen(false);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
let's implement this part:
![image](https://github.com/mate-academy/react_autocomplete/assets/86923089/0f66f200-3837-405c-a748-4b324a8add0a)

feel free to ask if you need help :slightly_smiling_face: